pkg/driver/node.go

```diff
@@ -213,6 +222,15 @@ func (d *Driver) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublish
 		return &csi.NodeUnpublishVolumeResponse{}, nil
 	}
 
+	deleteCount, err := decrementMapCounter(volumeContentDeleteTracker, req.GetVolumeId())
+	if err != nil {
+		klog.Errorf("Failed to check if volume content should be deleted: %v", err)
+	}
+
+	if deleteCount == 0 {
+		deleteDirectoryContent(req.GetTargetPath())
```

Comment:
why is this needed, seems a bit dangerous!

Reply:
Say I NodePublish the same volume 3x to different paths. driver restarts. I NodePublish 1 more time. I NodeUnpublish it. Won't it delete everything even though there are still 3 other consumers?

Reply:
There is currently no way to delete a root directory created by access points and the driver will delete only the access point inside `DeleteVolume`.
DeleteContent option will allow users/administrators to specify if they want to delete the data behind the root directory when they want to delete a volume. If the option is set, the driver will delete the data prior to unmount inside NodeUnpublish

This counter specifically tracks the if an access point (volume Handle [fs_id]::[ap_id]) is mounted in more than one volume, lets say Volume A and B. If A is deleted and B is still in use, driver doesn't delete the data till B is deleted.

Reply:
> Say I NodePublish the same volume 3x to different paths. driver restarts. I NodePublish 1 more time. I NodeUnpublish it. Won't it delete everything even though there are still 3 other consumers?

Ah good catch on this one! I did not think of the case where the driver restarts

Reply:
Yea, I'm more comfortable of doing this *only if* `volumeDeleteContent` is set explicitly, rather than relying on a map whose initial state would delete content. 

I'm surprised NodeUnpublishVolume request doesn't expose VolumeContext. But since we cache the result in `volumeDeleteContent`, we should explicitly check if it is still set to true before deletion.

Reply:
Thinking about this, caching this option is a bad idea. I'll get rid of this and provide users another way to retain their data.